report/report.md

```diff
@@ -0,0 +1,898 @@
+coefficient vector of $p(X)$, using the IPA. However, since the the vector
+$\vec{z}$ is not private, and has a certain structure, we can split the
+verification algorithm in two: $\PCDLSuccinctCheck$ and $\PCDLCheck$. Using
+the $\PCDLSuccinctCheck$ we can accumulate $n$ instances, and only perform
+the expensive linear check at the end of accumulation.
+
+In the [2020 paper _"Proof-Carrying Data from Accumulation
+Schemes"_](https://eprint.iacr.org/2020/499.pdf), that this project
+heavily relies on, the authors presented a generalized version of the
+previous accumulation structure of Halo that they coined _Accumulation
+Schemes_. Simply put, given a predicate $\Phi: X \to \Bb$,
+an accumulation scheme consists of the following functions:
+
+- $\ASProver(q_i: X, acc_i: \Acc) \to \Acc$
+
+    The prover accumulates the instance $q_i$ into the previous accumulator
+    $acc_{i-1}$ into the new accumulator $acc_i$.
+
+- $\ASVerifier(q_i: X, acc_i: \Acc, acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+    The verifier checks that the instance $q_i$ was correctly accumulated
+    into the previous accumulator $acc_{i-1}$ to form the new accumulator
+    $acc_i$.
+
+- $\ASDecider(acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+    The decider performs a single check that simultaneously ensures that
+    _every_ previous instance-proof pair accumulated in $acc_{i+1}$ verifies
+    assuming the $\ASVerifier$ has accepted that every previous accumulator
+    correctly accumulates $q_i$.
+
+We define completeness and soundness for the Accumulation Scheme:
+
+- **Completeness:** For all accumulators $acc_i$ and predicate inputs $q \in X$,
+  if $\top = \ASDecider(acc) = \Phi(q)$, then for $\ASProver(q, acc_i)
+  = acc_{i+1}$ it holds that $\top = \ASVerifier(acc_i, q, acc_{i+1}) =
+  \ASDecider(acc_{i+1})$.
+
+- **Soundness:** For all efficiently-generated accumulators $acc_i, acc_{i+1}
+  \in \Acc$ and predicate inputs $q \in X$, if $\top = \ASDecider(acc_{i+1}) =
+  \ASVerifier(q, acc_i, acc_{i+1})$ then, with all but negligible probability,
+  $\top = \Phi(q) = \ASDecider(acc_i)$.
+
+### IVC from Accumulation Schemes
+
+If the $\ASVerifier$ runtime is sub-linear and we have a $\SNARKVerifierSlow$
+that may run in linear time, then we can create an IVC scheme:
+
+- $\IVCProver(z_i: \textbf{Instance}, \pi_i: \textbf{Proof}, acc_i: \Acc) \to (\textbf{Proof}, \Acc)$
+
+  - Accumulates $(z_i, \pi_i)$ with $acc_i$ to obtain a new accumulator $acc_{i+1}$.
+  - Then generates a SNARK proof $\pi_{i+1}$ of the following claim, expressed as a circuit $R$:
+      $$"z_{i+1} = F(z_i) \; \land \; (i = 0 \lor \exists \, \pi_i, \, acc_i \text{ s.t. } \ASVerifier((z_i, \pi_i), acc_i, acc_{i+1}) = \top)"$$
+  - Output $(\pi_{i+1}, acc_{i+1})$
+
+- $\IVCVerifier(\pi_{i+1}: \textbf{Proof}, acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+  Checks the proof: $\top \meq \SNARKVerifierSlow(\pi_{i+1}) \meq \ASDecider(acc_{i+1})$
```

Comment:
SNARK verifier should take as input acc_{i+1} as well; i.e., it should be \SNARKVerifierSlow(\pi_{i+1}, acc_{i+1})

Reply:
From the paper:

> Given such an accumulation scheme, we can construct IVC as follows. Given a previous instance $z_i$, proof $\pi_i$, and accumulator $\text{acc}_i$, the IVC prover first accumulates $(z_i, \pi_i)$ with $\text{acc}_i$ to obtain a new accumulator $\text{acc}_{i+1}$. The IVC prover also generates a SNARK proof $\pi_{i+1}$ of the claim: "$z_{i+1} = F(z_i)$, and there exist a proof $\pi_i$ and an accumulator $\text{acc}_i$ such that the accumulation verifier accepts $((z_i, \pi_i), \text{acc}_i, \text{acc}_{i+1})$", expressed as a circuit $R$. The final IVC proof then consists of $(\pi_T, \text{acc}_T)$. _The IVC verifier checks such a proof **by running the SNARK verifier on $\pi_T$** and the accumulation scheme decider on $\text{acc}_T$._

Is the above wrong, or have I misunderstood it?